### PR TITLE
Update db scripts with auditing etc

### PR DESCRIPTION
**Pull request by Quiescent**
Augment the two new db scripts with auditing and dry run capabilities in preparation for the production run.

### CODE REVIEW

**Review by jeanawhou — APPROVED** [review]
LGTM :)

File: lib/plottr_firebase/migrations/copy-known-file-information-to-authorisations.js [hunk 1]
```diff
@@ -23,14 +25,92 @@ if (!admin.apps.length) {
...
+            console.log('User exists.  Executing script.')
+            const userIds = userId ? Promise.resolve([userId]) : allUserIds()
+            userIds
+              .then((uids) => {
+                console.log(`Will run script for users with ids: ${JSON.stringify(uids, null, 2)}`)
+                return sequencePromises(
+                  uids.map((uid) => {
+                    return () => {
+                      console.log(`> ${uid}`)
+                      return markDeletedAuthorisedDocumentsAsDeleted(
```
Comment (cameronsutter) [inline lib/plottr_firebase/migrations/copy-known-file-information-to-authorisations.js:56]: I think this migration is doing the wrong thing. The name of the file suggests it should be the  copy-known-file-info-to-authorisations one

File: lib/plottr_firebase/migrations/mark-deleted-authorised-documents.js [hunk 2]
```diff
@@ -60,26 +142,32 @@ const deletedFiles = (userId) => {
...
+  })
 }
 
-const markDeletedAuthorisedDocumentsAsDeleted = (userId) => {
+const markDeletedAuthorisedDocumentsAsDeleted = (userId, executingUserId, readOnly) => {
   const database = admin.firestore()
+  const runTransactionWithAuditing = logChange(readOnly)(
+    database,
+    executingUserId,
+    'copy-known-file-information-to-authorisations.js',
```
Comment (cameronsutter) [inline lib/plottr_firebase/migrations/mark-deleted-authorised-documents.js:168]: i think this line should be something to do with this migration, right? Something along the lines of 'mark-deleted-authorised-documents-as-deleted`

File: lib/plottr_firebase/migrations/copy-known-file-information-to-authorisations.js (hunk 1 above)
Comment (Quiescent) [reply lib/plottr_firebase/migrations/copy-known-file-information-to-authorisations.js:56]: Oops.  Copy-paste error.
Only the name is wrong :sweat_smile:

File: lib/plottr_firebase/migrations/mark-deleted-authorised-documents.js (hunk 2 above)
Comment (Quiescent) [reply lib/plottr_firebase/migrations/mark-deleted-authorised-documents.js:168]: Same thing here... thanks for spotting these! :)

**Pull request merged**